src/components/RichTextEditor/Plugins/BreakAfterPlugin/BreakAfterPlugin.ts

```diff
@@ -30,31 +35,101 @@ export class BreakAfterPlugin extends Plugin {
 }
 
 // This is adapted from packages/editor/break/src/soft-break/onKeyDownSoftBreak.ts
-const onKeyDownBreakAfter =
-    (editor: any, { options: { rules = [] } }): KeyboardHandlerReturnType =>
-    (event) => {
+const OnKeyDownBreakAfter = (editor: any): KeyboardHandlerReturnType => {
+    const { style } = useRichTextEditorContext();
+    const columns = Number(style?.columns) ?? 1;
+
+    return (event) => {
         const isActive = someNode(editor, { match: { breakAfterColumn: true } });
         const entry = getBlockAbove(editor);
 
         if (!entry) {
             return;
         }
 
-        for (const { hotkey, query } of rules) {
-            if (isHotkey(hotkey, event as any) && queryNode(entry, query)) {
-                getPreventDefaultHandler(setBreakAfter, editor, {
-                    value: !isActive,
-                    key: KEY_ELEMENT_BREAK_AFTER,
-                })(event);
-            }
+        switch (event.key) {
+            case 'Enter':
+                handleEnterKeyEvent(editor, columns, isActive, event);
+                break;
+
+            case 'Backspace':
+                handleBackSpaceKeyEvent(editor, isActive, event);
+                break;
+
+            case 'Delete':
+                const anchor = editor.selection.anchor;
+                if (anchor.offset === getNode(editor, anchor.path).text.length && isActive) {
+                    getPreventDefaultHandler(setBreakAfter, editor, {
+                        value: false,
+                    })(event);
+                }
+                const pointAfter = getPointAfter(editor, editor.selection);
+                const elementAfterIsActive = someNode(editor, {
```

Comment:
isn't this always the case (isn't this condition obsolete)?

Reply:
Oh good catch! Removed it.